webkit2/tests/tests.lisp

```diff
@@ -0,0 +1,418 @@
+         (is (eq :true %result%)))
+    (setf webkit::*js-true-value* t)))
+
+;;; Numbers
+
+(def-test integers (:suite js-tests)
+  (with-js-transform-result "0" (%result%)
+    (is (= 0 %result%)))
+  (with-js-transform-result "8" (%result%)
+    (is (= 8 %result%)))
+  (with-js-transform-result "-6" (%result%)
+    (is (= -6 %result%)))
+  (with-js-transform-result "Math.pow(10, 100)" (%result%)
+    (is (= *googol* %result%)))
+  (with-js-transform-result "-Math.pow(10, 100)" (%result%)
+    (is (= (- *googol*) %result%))))
+
+(def-test special-numbers (:suite js-tests)
+  (with-js-transform-result "NaN" (%result%)
+    (is (float-features:float-nan-p %result%)))
+  (with-js-transform-result "Infinity" (%result%)
+    (is (float-features:float-infinity-p %result%))
+    (is (equal 1 (truncate (float-sign %result% 1.0)))))
+  (with-js-transform-result "-Infinity" (%result%)
+    (is (float-features:float-infinity-p %result%))
+    (is (equal -1 (truncate (float-sign %result% 1.0))))))
+
+(def-test fractional-number (:suite js-tests)
+  (with-js-transform-result "5.3" (%result%)
+    (is (= 5.3 (coerce %result% 'single-float))))
+  (with-js-transform-result "-34435.3044" (%result%)
+    (is (= -34435.3044 (coerce %result% 'single-float)))))
+
+(def-test periodic-number (:suite js-tests)
+  (with-js-transform-result "var num = 5/3; num" (%result%)
+    (is (equal 1.6666666666666667 (coerce %result% 'single-float)))))
+
+;;; Strings
+
+(def-test simple-string (:suite js-tests)
+  (with-js-transform-result "\"hello\"" (%result%)
+    (is (equal "hello" %result%))))
+
+(def-test escaped-string (:suite js-tests)
+  (with-js-transform-result "\"hello\\nthere\"" (%result%)
+    (is (equal "hello
+there" %result%))))
+
+(def-test templated-string (:suite js-tests)
+  (with-js-transform-result "var num = 5; `${num} + ${num} = ${10}`" (%result%)
+    (is (equal "5 + 5 = 10" %result%))))
+
+(def-test concatenated-string (:suite js-tests)
+  (with-js-transform-result "\"hello \" + \"there!\"" (%result%)
+    (is (equal "hello there!" %result%))))
+
+;;; Arrays
+
+(def-test simple-array (:suite js-tests)
+  (with-js-transform-result "[1, 2, 3, 4, 5]" (%result%)
+    (is (equal (list 1 2 3 4 5) %result%)))
+  (with-js-transform-result "[\"h\", \"e\", \"l\", \"l\", \"o\"]" (%result%)
+    (is (equal (list "h" "e" "l" "l" "o") %result%)))
+  (with-js-transform-result "[true, false, true, true, false]" (%result%)
+    (is (equal (list t nil t t nil) %result%))))
+
+(def-test everything-array (:suite js-tests)
+  (setf webkit:*js-object-type* :alist)
+  (unwind-protect
+       (with-js-transform-result "[true, false, undefined, null, 100000, \"hello\", {one: 1}]" (%result%)
+         (is (equal (list t nil :undefined :null 100000 "hello" '(("one" 1))) %result%)))
+    (setf webkit:*js-object-type* :hash-table)))
+
+(def-test everything-array-vector (:suite js-tests)
+  (setf webkit:*js-array-type* :vector
+        webkit:*js-object-type* :alist)
```

Comment:
These `unwind-protect` are the unfortunate consequence that we cannot bind the dynamic variables properly, right?

What about making an (unexported) helper macro for now?  Maybe just for testing.